ath79: Add green power LED to Airtight C-75

svenschwermer opened the pull request:
In addition to the missing green LED definition, the polarity of the
amber power LED was incorrect which is fixed here.

Signed-off-by: Sven Schwermer <sven@svenschwermer.de>

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

mans0n commented:
AFAIK the current `get_dt_led()` does not support the new color/function scheme, so it will fail to find the boot led. (It can be fixed by chunkeey's [patch](https://git.openwrt.org/?p=openwrt/staging/chunkeey.git;a=commitdiff;h=a956f3880d8cdb88dcb9e3d26831235fbaf61abe))
For now you'll need to keep the led labels.

ynezz commented:
@mans0n Well, I've noticed, that it's something new, so in order to check it, I've just `git grep LED_FUNCTION_POWER` and got like more then 20 results, so I've thought, that it somehow works :-) @chunkeey what is your plan with that fix of yours?

svenschwermer commented:
@mas0n thanks for your comment. I'm happy to amend this commit to add back the labels.
@ynezz please advise which way forward you would prefer.

mans0n commented:
The color/function scheme does work, but not with led aliases. So to fix it, just adding back the labels will do the job. (No need to remove color/function properties)

Suggestion:
```diff
		led_power_amber: power-amber {
+			label = "amber:power";
			color = <LED_COLOR_ID_AMBER>;
			function = LED_FUNCTION_POWER;
			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
		};

		led_power_green: power-green {
+			label = "green:power";
			color = <LED_COLOR_ID_GREEN>;
			function = LED_FUNCTION_POWER;
			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
		};
```

svenschwermer committed:
ath79: Add LED labels for Airtight C-75

Keep labels since OpenWrt userland tooling (get_dt_led) depends on them
to find the LED instances referenced by the led-* aliases.

The label for the amber power LED was removed in 4eefdc7adb.

Signed-off-by: Sven Schwermer <sven@svenschwermer.de>

mans0n commented:
I don't think using a string literal for `function` is a good solution as it does not match upstream's recommendation. Please leave it as is, or use `function = LED_FUNCTION_WLAN` and `function-enumerator = <1>`. (I prefer the former, label is still good for us.)

And the commit title should be something like `ath79: add LED labels for Airtight C-75`.

chunkeey commented:
About that get_leds_dt() patch, it's probably DOA. I came across @rmilecki solution for tackling this in bcm53xx (because most/all boards are upstream in this target): 

https://github.com/openwrt/openwrt/blob/master/target/linux/bcm53xx/base-files/etc/diag.sh#L5-L24

That said, this requires major dts (re-)work for existing targets :frowning:

svenschwermer commented:
I just tested this second commit. Now, everything appears to work. Thanks for looking into this, everyone :)

mans0n commented:
> About that get_leds_dt() patch, it's probably DOA. I came across @rmilecki solution for tackling this in bcm53xx (because most/all boards are upstream in this target)

Err, why? bcm53xx diag.sh is functionally inferior to generic diag.sh. It's less flexible (why favor "power" and not "status" or "cpu"?) and cannot deal with multiple status LEDs (i.e. boot/running/failsafe/upgrade).

mans0n commented:
Thanks! Merged your patch as 591a2b952541.
